Userland/DynamicLoader/main.cpp

```diff
@@ -36,18 +38,75 @@ static void perform_self_relocations(auxv_t* auxvp)
         exit(1);
 }
 
-static void display_help()
+static ErrorOr<int> open_executable(StringView path)
 {
-    char const message[] =
-        R"(You have invoked `Loader.so'. This is the helper program for programs that
-use shared libraries. Special directives embedded in executables tell the
-kernel to load this program.
-
-This helper program loads the shared libraries needed by the program,
-prepares the program to run, and runs it. You do not need to invoke
-this helper program directly. If you still want to run a program with the loader,
-run: /usr/lib/Loader.so [ELF_BINARY])";
-    fprintf(stderr, "%s", message);
+    int rc = open(path.characters_without_null_termination(), O_RDONLY | O_EXEC);
+    if (rc < 0)
+        return Error::from_errno(errno);
+    int checked_fd = rc;
+    ArmedScopeGuard close_on_failure([checked_fd] {
+        close(checked_fd);
+    });
+
+    struct stat executable_stat = {};
+    rc = fstat(checked_fd, &executable_stat);
+    if (rc < 0)
+        return Error::from_errno(errno);
+    if (!S_ISREG(executable_stat.st_mode)) {
+        if (S_ISDIR(executable_stat.st_mode))
+            return Error::from_errno(EISDIR);
+        return Error::from_errno(EINVAL);
+    }
+
+    close_on_failure.disarm();
+    return checked_fd;
+}
+
+static int _main(int argc, char** argv, char** envp, bool is_secure)
+{
+    tzset();
```

Comment:
Why?

Reply:
I thought that I might replicate the usual main function call flow, so it is there, why not do this also here for the sake of completeness?